specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1007 @@
+unpopulated. Once created, the `spec.channelTemplate` field MUST be immutable;
+the Channel MUST be deleted and re-created to change the `spec.channelTemplate`.
+This pattern is chosen to make it clear that changing `spec.channelTemplate` is
+not an atomic operation and that any implementation would be likely to result in
+event loss during the transition.
+
+### Subscription Lifecycle
+
+A Subscription MAY be created before the referenced Channel indicated by its
+`spec.channel` field. The `spec.channel` object reference MAY refer to either an
+`messaging.knative.dev/v1` Channel resource, or another resource which meets the
+`spec.subscribers` and `spec.delivery` required elements in the Channelable duck
+type. If the referenced `spec.channel` does not currently exist or its `Ready`
+condition is not `true`, then the Subscription's `Ready` condition MUST NOT be
+`true`, and the reason SHOULD indicate that the corresponding Channel is missing
+or not ready.
+
+The Subscription MUST also set the `status.physicalSubscription` URIs by
+resolving the `spec.subscriber`, `spec.reply`, and
+`spec.delivery.deadLetterSink` as described in
+[Addressable resolution](#addressable-resolution) before setting the `Ready`
+condition to `true`. If any of the addressable fields fails resolution, the
+Subscription MUST set the `Ready` condition to `false`, and at least one
+condition MUST indicate the reason for the error. The Subscription SHOULD also
+set `status.physicalSubscription` URIs to the empty string if the corresponding
+`spec` reference cannot be resolved.
+
+At least one of `spec.subscriber` and `spec.reply` MUST be set; if only
+`spec.reply` is set, the behavior is equivalent to setting `spec.subscriber`
+except that the Channel SHOULD NOT advertise the ability to process replies
+during the delivery.
+
+Once created, the Subscription's `spec.channel` MUST NOT permit updates; to
+change the `spec.channel`, the Subscription can be deleted and re-created. This
+pattern is chosen to make it clear that changing the `spec.channel` is not an
+atomic operation, as it might span multiple storage systems. Changes to
+`spec.subscriber`, `spec.reply`, `spec.delivery` and other fields SHOULD be
+permitted, as these could occur within a single storage system.
+
+When a Subscription becomes associated with a Channel (either due to creating
+the Subscription or the Channel), the Subscription MUST only set the `Ready`
+condition to `true` after the Channel has been configured to send all future
+events to the Subscriptions `spec.subscriber`. The Channel MAY send some events
+to the Subscription before prior to the Subscription's `Ready` condition being
+set to `true`. When a Subscription is deleted, the Channel MAY send some
+additional events to the Subscription's `spec.subscriber` after the deletion.
+
+<!--
+TODO: channel-compatible CRDs (Channelable)
+-->
+
+### Addressable Resolution
+
+Both Trigger and Subscription have OPTIONAL object references (`ref` in
```

Comment:
It might be better to make this more generic since there might be other types that are addressable and I think this section should cover all of them. So, perhaps talk about it like this:
The Event specifications refer to objects that are "Addressable", this means that it MUST be possible to determine a URL for those object references. There is a partial schema ("duck type") that these objects MAY support which allows for easy discovery of its URL. ...then talk about how to combine ref+url and k8s services....

There MAY also be an out-of-band mechanism by which an implementation can determine an object's URL if it is not duck-type addressable.

(see next comment too - since i think they go together)

Reply:
Yup- don't care too much about the wording as long as there isn't text like "the objRef MUST be duck-type addressable" and allows for other mechanisms to be defined - which I think there is today.